fixing Operator/Authority choice in line

ue71603 opened the pull request:
like #613  #341 but against master

skinkie commented on xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:
```diff
@@ -406,16 +406,19 @@ Rail transport, Roads and Road transport
+			<xsd:element ref="AuthorityRef" minOccurs="0">
 				<xsd:annotation>
-					<xsd:documentation>Primary OPERATOR of LINE.</xsd:documentation>
+					<xsd:documentation>A line can only have no Authority when this information is provided in a ResponsibilitySet.</xsd:documentation>
```
I dislike this documenation sentense.

ue71603 replied:
updated

skinkie commented on xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:
```diff
@@ -406,16 +406,19 @@ Rail transport, Roads and Road transport
+			</xsd:element>
+			<xsd:element ref="OperatorRef" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Operators should be set, even when the same as the Authority. Can be omitted when provided via ResponsibilitySet</xsd:documentation>
```
No, don't allow it to be omitted...

ue71603 replied:
changed

skinkie commented on xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:
```diff
@@ -406,16 +406,19 @@ Rail transport, Roads and Road transport
+					<xsd:documentation>Operators should be set, even when the same as the Authority. Can be omitted when provided via ResponsibilitySet</xsd:documentation>
+				</xsd:annotation>
+			</xsd:element>
 			<xsd:element name="additionalOperators" type="transportOrganisationRefs_RelStructure" minOccurs="0">
```
I think additionalOperators where added when we had a choice of Authority or Operator. Should we discuss this too?

ue71603 replied:
my comment ok?

ue71603 commented:
@skinkie  I think one still can have additional operators.  The main case is 1-1, but there are others. Or do you want to change it?

ue71603 commented on xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:
```diff
@@ -406,16 +406,19 @@ Rail transport, Roads and Road transport
+			<xsd:element ref="AuthorityRef" minOccurs="0">
 				<xsd:annotation>
-					<xsd:documentation>Primary OPERATOR of LINE.</xsd:documentation>
+					<xsd:documentation>A line can only have no Authority when this information is provided in a ResponsibilitySet.</xsd:documentation>
```
```suggestion
					<xsd:documentation>A line should alway have an Authority, even when the information is provided in a ResponsibilitySet.</xsd:documentation>
```

trurlurl replied:
```suggestion
					<xsd:documentation>A line should always have an Authority, even when the information is provided in a ResponsibilitySet.</xsd:documentation>
```

ue71603 commented on xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:
```diff
@@ -406,16 +406,19 @@ Rail transport, Roads and Road transport
-				<xsd:element ref="OperatorRef"/>
-			</xsd:choice>
+			</xsd:element>
+			<xsd:element ref="OperatorRef" minOccurs="0">
```
```suggestion
```

ue71603 commented:
fixed a bug I introduced
@skinkie  @Aurige  @duex 
it is currently possible to provide neither Authorithy nor Operator (and do everything in ResponsibilitySets).
If you don't want that, then intervene (btw this is what SBB is doing)

Robbendebiene commented on xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:
```diff
@@ -406,16 +406,19 @@ Rail transport, Roads and Road transport
+			<xsd:element ref="AuthorityRef" minOccurs="0">
 				<xsd:annotation>
-					<xsd:documentation>Primary OPERATOR of LINE.</xsd:documentation>
+					<xsd:documentation>A line should alway have an Authority, even when the information is provided in a ResponsibilitySet.</xsd:documentation>
```
```suggestion
					<xsd:documentation>A line should always have an Authority, even when the information is provided in a ResponsibilitySet.</xsd:documentation>
```

skinkie commented on xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:
```diff
@@ -406,16 +406,19 @@ Rail transport, Roads and Road transport
+			<xsd:element ref="AuthorityRef" minOccurs="0">
 				<xsd:annotation>
-					<xsd:documentation>Primary OPERATOR of LINE.</xsd:documentation>
+					<xsd:documentation>A line should always have an Authority, even when the information is provided in a ResponsibilitySet.</xsd:documentation>
```
Do you want to write AUTHORITY here or isn't that a Transmodel concept?

ue71603 replied:
changed

skinkie commented on xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:
```diff
@@ -406,16 +406,19 @@ Rail transport, Roads and Road transport
+			</xsd:element>
+			<xsd:element ref="OperatorRef" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Operators should be set, even when the same as the Authority. In some cases Operator/Authority are managed through ResponsibilitySet. However, for compatibility OperatorRef and AuthorityRef still should be filled in.</xsd:documentation>
```
```suggestion
					<xsd:documentation>An OPERATOR should be set, even when the same as the AUTHORITY. In some cases OPERATOR or AUTHORITY are managed through a ResponsibilitySet. However, for compatibility OperatorRef and AuthorityRef still should be filled in.</xsd:documentation>
```
